apps/hash-frontend/src/components/grid/grid.tsx

```diff
@@ -469,6 +479,11 @@ export const Grid = <T extends Row & { rowId: string }>({
     [columns, openFilterColumnKey],
   );
 
+  if (currentlyDisplayedRowsRef && sortedAndFilteredRows) {
+    // eslint-disable-next-line no-param-reassign
+    currentlyDisplayedRowsRef.current = sortedAndFilteredRows;
+  }
```

Comment:
This looked a bit out of place to me... my instinct would be to lift the `sortedAndFilteredRows` state out of the component (as `selectedRows` already is, for example). 

Doing this via a ref takes a bit longer to read, because it prompts the question of why it's done like this. One explanation is to prevent the parent component from re-rendering when the sorted and filtered rows change. Another is that moving it up would also involve moving the column filter state etc up and until we need filtering/sorting in more places, it's not worth it. It might be worth adding a comment explaining why it's done like this.

Reply:
I agree it's a bit odd, but the reasons you mentioned I wasn't sure how to best approach exposing this without moving the filtering logic out of the `Grid` component, which I'm not too sure about. I've added a comment in https://github.com/hashintel/hash/pull/3940/commits/c5991703dd91bb3be0697f6378adfc9dea9adf77 to make this a bit clearer for now.